### PR TITLE
Options parameter and new overlay

### PR DESCRIPTION
**Pull request by Stending**
## Description
Based on the refactor_options branch #45 by @nleclerc.
The overlay was improved, it is now a bit more organized and flexible.

The devices icon is now disabled by default, but, only on android, you can display it by adding a boolean parameter "devicesIconVisible" to the options object, and set it to true.
It doesn't work on iOS because there is no proper way to send overlay options to the ImageTargetsViewController, for now. Might be a future pull request.

An android InApp close button is now available. It is disabled by default to, but you can enable it with the parameter showAndroidCloseButton to true.

## Motivation and Context
A InApp close button on android can be useful for app disabling the back button, like our.
A more stable and customizable overlay is useful to. It fits more with our app, and is a bit more generic.

## How Has This Been Tested?
Tested on an iPod Touch and a Galaxy S2.

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [x] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] My code follows the code style of this project.
- [] My change requires a change to the documentation.
- [x] I have read the **CONTRIBUTING** document.

### CODE REVIEW

**Comment by mattrayner** [comment]
Any particular reason why you changed the iOS close button as well as adding an Android one?

**Comment by mattrayner** [comment]
My main concern with that button image is, if you point your camera at a grey image, the close button is completely obscured. This was the reasoning behind the original black/white image.

My concern is that merging this will lower the usability of the plugin

**Comment by Stending** [comment]
We tested it a lot, the cross is light grey, almost white, and the bar is opaque black, you'll see the button on a dark picture, and on a clear picture.

**Comment by mattrayner** [comment]
Okay cool, if you rebase I'll look at merging this in 👍

**Comment by Stending** [comment]
I didn't rebase, but I just merged with the actual master, it's easier for both of us!

**Comment by mattrayner** [comment]
That wont work though as just merging doesn't fix conflicts - I'll be unable to merge until you do a rebase

**Comment by Stending** [comment]
I don't understand, I resolved the conflicts during my merge. What's your issue?

**Comment by mattrayner** [comment]
Looks like GitHub was delayed - after you pushed, I was still unable to merge - GitHub was reporting a conflict 👍

File: www/VuforiaPlugin.js [hunk 1]
```diff
@@ -9,22 +9,32 @@ var VuforiaPlugin = {
...
    * @param {function|null} errorCallback A callback for when an error occurs. Could include device not having a camera,
    *                                      or invalid Vuforia key. Passes an error string with more information.
    */
-  startVuforia: function(imageFile ,imageTargets, overlayCopy, vuforiaLicense, imageFoundCallback, errorCallback){
+  startVuforia: function(options, imageFoundCallback, errorCallback){
+    var imageFile = options.databaseXmlFile;
+    var imageTargets = options.targetList;
+    var overlayCopy = options.overlayMessage;
+    var vuforiaLicense = options.vuforiaLicense;
+    var showAndroidCloseButton = options.showAndroidCloseButton?true:false;
```
Comment (mattrayner) [inline www/VuforiaPlugin.js]: Can we add spaces here to keep in line with code style?

File: README.md [hunk 2]
```diff
@@ -38,11 +37,15 @@ Cordova-Plugin-Vuforia comes with two JavaScript methods, `startVuforia`, and `s
 From within your JavaScript file, add the following to launch the [Vuforia][vuforia] session.
 
 ```javascript
+var options = {
+  databaseXmlFile: 'PluginTest.xml',
+  targetList: [ 'logo', 'iceland', 'canterbury-grass', 'brick-lane' ],
+  overlayMessage: 'Point your camera at a test image...',
+  vuforiaLicense: 'YOUR_VUFORIA_KEY'
```
Comment (mattrayner) [inline README.md]: Should we update this to include `showAndroidClose`?

**Comment by mattrayner** [comment]
That's awesome! Thanks, I have just a couple of comments then we are all good 👍

File: README.md (hunk 2 above)
Comment (Stending) [inline README.md]: Absolutely!

File: www/VuforiaPlugin.js (hunk 1 above)
Comment (Stending) [inline www/VuforiaPlugin.js]: I don't see any problem with that, that's my mistake.

**Pull request merged**